process/block/baseProcess.go

```diff
@@ -979,11 +979,6 @@ func (bp *baseProcessor) updateStateStorage(
 		return
 	}
 
-	if finalHeader.IsStartOfEpochBlock() {
```

Comment:
If there is a snapshot twice at the same root hash, the second snapshot will not be completed, because there already is a snapshot at that root hash, so there should be no problem.

Reply:
Also you can create this map as following: map[rootHash][header nonce], and header nonce would be the nonce of the currentHeader param from func (sp *shardProcessor) updateState(headers []data.HeaderHandler, currentHeader *block.Header). In this way you can clean this map for nonces behind final  on CommitBlock method

Reply:
The problem in snapShotEpochStartFromMeta appears also when a epoch start meta block appears in more shard headers which is possible as it could be fully processed in more iterations. So the above proposed protection will solve also this situation.

Reply:
In case of shardBlockProcessor the nonce of this local map proposed above should be nonce of the start of epoch meta block and should be clean on CommitBlock if it is behind lastCrossNotarizedHeader nonce

Reply:
In cse of metaBlockProcessor this nonce should be the nonce of the lastMetaBlock parameter from the updateState method and also should be clean on CommitBlock if it is behind highet final block nonce

Reply:
Ok, cool, I have not seen the comment of @BeniaminDrasovean so forget about all these messages in this case :))